DragaliaAPI/Services/Game/QuestTreasureService.cs

```diff
@@ -0,0 +1,82 @@
+using DragaliaAPI.Database;
+using DragaliaAPI.Database.Entities;
+using DragaliaAPI.Database.Repositories;
+using DragaliaAPI.Features.Reward;
+using DragaliaAPI.Shared.Definitions.Enums;
+using DragaliaAPI.Shared.MasterAsset;
+using DragaliaAPI.Shared.MasterAsset.Models;
+using DragaliaAPI.Shared.PlayerDetails;
+using DragaliaAPI.Models.Generated;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DragaliaAPI.Services.Game;
+
+public class QuestTreasureService(
+    IRewardService rewardService,
+    IUserDataRepository userDataRepository,
+    IPlayerIdentityService playerIdentityService,
+    IUpdateDataService updateDataService,
+    ApiContext apiContext
+) : IQuestTreasureService
+{
+    public async Task<QuestOpenTreasureData> DoOpenTreasure(QuestOpenTreasureRequest request)
+    {
+        QuestTreasureData questTreasureData = MasterAsset.QuestTreasureData[
+            request.quest_treasure_id
+        ];
+
+        await rewardService.GrantReward(
+            new Entity(
+                questTreasureData.EntityType,
+                questTreasureData.EntityId,
+                questTreasureData.EntityQuantity
+            )
+        );
```

Comment:
What does this do when EntityType/EntityId/EntityQuantity are 0, like for `102302`?

Reply:
zero is Entity Type None so rewardService isn't used in that case.